### PR TITLE
Refractored Pose Estimation Client to wrap the different parameters into dataclasses. 

### PR DESCRIPTION
**Pull request by Zaid-Waghoo**

### CODE REVIEW

File: mbodied/agents/sense/object_pose_estimator_3d.py [hunk 1]
```diff
@@ -60,39 +62,70 @@ def act(
         self,
         rgb_image_path: str,
         depth_image_path: str,
-        camera_intrinsics: List[float] | np.ndarray,
-        distortion_coeffs: List[float] | None = None,
-        aruco_pose_world_frame: Pose6D | None = None,
-        object_classes: List[str] | None = None,
-        confidence_threshold: float | None = None,
+        camera_intrinsics: IntrinsicParameters = None,
```
Comment (sebbyjp) [inline mbodied/agents/sense/object_pose_estimator_3d.py]: if required, remove defaults, threshold set to default value,

**Review by sebbyjp — APPROVED** [review]

File: mbodied/agents/sense/camera_params.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+class CameraParameters(Sample):
+    """Model for Camera Parameters."""
+    intrinsic: IntrinsicParameters = Field(
+        default_factory=IntrinsicParameters, description="Intrinsic parameters of the camera")
+    distortion: DistortionParameters = Field(
+        default_factory=DistortionParameters, description="Distortion parameters of the camera")
+    extrinsic: ExtrinsicParameters = Field(
+        default_factory=ExtrinsicParameters, description="Extrinsic parameters of the camera")
+    depth_scale: float = 1.0
```
Comment (nqyy) [inline mbodied/agents/sense/camera_params.py:35]: Format with ruff pls. No new line here.

File: mbodied/types/sense/camera_params.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+class CameraParameters(Sample):
+    """Model for Camera Parameters."""
+    intrinsic: IntrinsicParameters = Field(
+        default_factory=IntrinsicParameters, description="Intrinsic parameters of the camera")
+    distortion: DistortionParameters = Field(
+        default_factory=DistortionParameters, description="Distortion parameters of the camera")
+    extrinsic: ExtrinsicParameters = Field(
+        default_factory=ExtrinsicParameters, description="Extrinsic parameters of the camera")
+    depth_scale: float = 1.0
```
Comment (nqyy) [inline mbodied/types/sense/camera_params.py:35]: Same here.

File: README.md [hunk 4]
```diff
@@ -64,11 +64,11 @@
     - [Notebooks](#notebooks)
   - [The Sample Class](#the-sample-class)
     - [💡 Did you know](#-did-you-know)
-  - [Building Blocks](#building-blocks)
       - [Creating a Sample](#creating-a-sample)
       - [Serialization and Deserialization with Pydantic](#serialization-and-deserialization-with-pydantic)
       - [Converting to Different Containers](#converting-to-different-containers)
       - [Gym Space Integration](#gym-space-integration)
+  - [Building Blocks](#building-blocks)
```
Comment (nqyy) [inline README.md:71]: Accidental change? Or rebase problem?

**Pull request closed without merging**